lib/openldap.c

```diff
@@ -582,6 +582,12 @@ static CURLcode oldap_connect(struct Curl_easy *data, bool *done)
 #ifdef USE_SSL
   if(conn->handler->flags & PROTOPT_SSL)
     return oldap_ssl_connect(data, OLDAP_SSL);
+  else {
+    CURLcode result = oldap_ssl_connect(data, OLDAP_SSL);
+
+    if(result)
+      return result;
+  }
```

Comment:
First, with this change it does the same thing in both branches.

Then: why call connect with `OLDAP_SSL` set if this handler doesn't use SSL ?

Reply:
Im not experienced with `ldap` module, so cannot be precise and this looks weird to me as well. In my opinion, it is just naming inconsistency because `oldap_ssl_connect` dispatchs to `Curl_conn_connect` and seemingly it understands whether the protocol wants SSL or not. Then, connects appropriately.

Reply:
If so, isn't this a slightly more covering fix with less repetition?
~~~diff
diff --git a/lib/openldap.c b/lib/openldap.c
index c9417fbe2..760bc6334 100644
--- a/lib/openldap.c
+++ b/lib/openldap.c
@@ -520,10 +520,11 @@ static CURLcode oldap_connect(struct Curl_easy *data, bool *done)
   struct connectdata *conn = data->conn;
   struct ldapconninfo *li;
   static const int version = LDAP_VERSION3;
   int rc;
   char *hosturl;
+  CURLcode result;
 #ifdef CURL_OPENLDAP_DEBUG
   static int do_trace = -1;
 #endif
 
   (void)done;
@@ -531,11 +532,10 @@ static CURLcode oldap_connect(struct Curl_easy *data, bool *done)
   DEBUGASSERT(!conn->proto.ldapc);
   li = calloc(1, sizeof(struct ldapconninfo));
   if(!li)
     return CURLE_OUT_OF_MEMORY;
   else {
-    CURLcode result;
     li->proto = ldap_pvt_url_scheme2proto(data->state.up.scheme);
     conn->proto.ldapc = li;
 
     /* Initialize the SASL storage */
     Curl_sasl_init(&li->sasl, data, &saslldap);
@@ -577,20 +577,26 @@ static CURLcode oldap_connect(struct Curl_easy *data, bool *done)
   ldap_set_option(li->ld, LDAP_OPT_PROTOCOL_VERSION, &version);
 
   /* Do not chase referrals. */
   ldap_set_option(li->ld, LDAP_OPT_REFERRALS, LDAP_OPT_OFF);
 
+  result = oldap_ssl_connect(data, OLDAP_SSL);
 #ifdef USE_SSL
   if(conn->handler->flags & PROTOPT_SSL)
-    return oldap_ssl_connect(data, OLDAP_SSL);
+    return result;
+  else if(result)
+    return result;
 
   if(data->set.use_ssl) {
-    CURLcode result = oldap_perform_starttls(data);
+    result = oldap_perform_starttls(data);
 
     if(!result || data->set.use_ssl != CURLUSESSL_TRY)
       return result;
   }
+#else
+  if(result)
+    return result;
 #endif
 
   if(li->sasl.prefmech != SASL_AUTH_NONE)
     return oldap_perform_mechs(data);

Reply:
Yes, it is better solution, let me apply the patch